### ISSUE TITLE
distro fails because it can't read /etc/image_version file

### ISSUE THREAD

**Issue by omarkohl** [nir0s/distro#162]
* Distro version: 1.0.2
* Python version: 3.4.3
* Operating system: Ubuntu 14.04

### Description:

On my system there is a file (out of my control) named /etc/image_version . This file is NOT world-readable. When using distro 1.0.2 in one of my virtualenvs (which I expect to be "isolated" from the system) distro will crash because it can't read said file.

I think the PermissionError should just be catched and ignored...

This issue was originally seen with pip (that vendors distro) and was reported here: https://github.com/pypa/pip/issues/4249

### What I've run:

```
testuser@host:~/tmp$ virtualenv -p python3 pip-test
Running virtualenv with interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in pip-test/bin/python3
Also creating executable in pip-test/bin/python
Installing setuptools, pip...done.
testuser@host:~/tmp$ source pip-test/bin/activate
(pip-test)testuser@host:~/tmp$ pip install distro
Downloading/unpacking distro
  Downloading distro-1.0.2-py2.py3-none-any.whl
Installing collected packages: distro
Successfully installed distro
Cleaning up...
(pip-test)testuser@host:~/tmp$ distro
Traceback (most recent call last):
  File "/home/testuser/tmp/pip-test/bin/distro", line 7, in <module>
    from distro import main
  File "/home/testuser/tmp/pip-test/lib/python3.4/site-packages/distro.py", line 1051, in <module>
    _distro = LinuxDistribution()
  File "/home/testuser/tmp/pip-test/lib/python3.4/site-packages/distro.py", line 596, in __init__
    self._distro_release_info = self._get_distro_release_info()
  File "/home/testuser/tmp/pip-test/lib/python3.4/site-packages/distro.py", line 996, in _get_distro_release_info
    distro_info = self._parse_distro_release_file(filepath)
  File "/home/testuser/tmp/pip-test/lib/python3.4/site-packages/distro.py", line 1016, in _parse_distro_release_file
    with open(filepath) as fp:
PermissionError: [Errno 13] Permission denied: '/etc/image_version'
```

**Comment by dstufft** [comment]
Would it be possible to get a release with the fix for this inside it?

**Comment by nir0s** [comment]
Sure. I can do one today.

On Mar 18, 2017 18:09, "Donald Stufft" <notifications@github.com> wrote:

Would it be possible to get a release with the fix for this inside it?

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/nir0s/distro/issues/162#issuecomment-287555902>, or mute
the thread

**Comment by nir0s** [comment]
@dstufft released v1.0.3 with the fix

**Comment by dstufft** [comment]
Thanks!
